src/ReplayTimeline.cc

```diff
@@ -218,6 +222,33 @@ ReplayTimeline::Mark ReplayTimeline::mark() {
   return result;
 }
 
+ReplayTimeline::Mark ReplayTimeline::recreate_mark_from_data(const MarkData& mark_data, ReplaySession::shr_ptr session) {
+  Mark result;
+  auto m = make_shared<InternalMark>(this, mark_data, session);
+  if (marks.empty()) {
+    marks = {};
+  }
+
+  if (marks.count(m->proto.key) == 0) {
+    marks[m->proto.key] = {};
+  }
+  auto& mark_vector = marks[m->proto.key];
```

Comment:
I think this & and the other `marks = {}` is a left over from when I began working on it, and I would get arbitrary segfaults because these were not initialized. `map[someKey]` should default construct, so I don't think I have to do either, really.

Will be fixed in coming commits